components/datadog/apps/dda/datadogagent.go

```diff
@@ -84,12 +84,17 @@ func K8sAppDefinition(e config.Env, kubeProvider *kubernetes.Provider, namespace
 		configureImagePullSecret(ddaConfig, imagePullSecret)
 	}
 
+	ddaName := "datadog-agent"
+	if e.PipelineID() != "" {
+		ddaName = fmt.Sprintf("datadog-agent-%s", e.PipelineID())
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
		ddaName = strings.Join([]string{ddaName, e.PipelineID()}, "-")
```